docs/validator.md

```diff
@@ -0,0 +1,261 @@
+
+JSON Schemas provide a clear way to indicate what is and is not valid within a given JSON document.  You might use JSON
+Schemas to validate a form's data on the client side before attempting to submit it to the server.  You might also
+validate data and report errors whenever a component's model changes.
+
+# How does it work?
+
+This package provides client and server-side components that use [AJV](https://github.com/epoberezkin/ajv) to validate
+supplied JSON data against a JSON Schema.  To use this on the client side, start with `gpii.schema.validator`.  To use
+it on the server side, start with `gpii.schema.validator.server`.
+
+The main work of the validator component is performed by its `validate(schemaKey, JSON)` invoker.  Here is an example
+of how the validator might be used from within node:
+
+    var fluid = require("infusion"");
+    var gpii  = fluid.registerNamespace("gpii");
+
+    require("gpii-json-schema");
+    var validator = gpii.schema.validator({
+        schemaContents: {
+          sample: { "type": "object", "properties": { "required": { "type": "boolean" } }, "required": ["required"]}
+        }
+    });
+
+    var errors = validator.validate("sample", { foo: "bar" });
+    if (errors) {
+        // Complain
+    }
+    else {
+        // Rejoice
+    }
+
+
+If there are no validation errors, we return `undefined`.  If there are validation errors, the raw output generated by
+AJV is transformed using the static function `sanitizeValidationErrors`.
+
+As an example, let's assume that you have a JSON payload like the following that you wish to validate:
+
+  ```
+  {
+    field1: "invalid",
+    category1: {
+      nestedField1: "invalid"
+    }
+  }
+  ```
+
+Assuming that each value was invalid, you might see validation output like the following:
+
+  ```
+  {
+    fieldErrors: {
+      field1: ["should NOT be shorter than 8 characters", "should match pattern \"[A-Z]+\""],
+      category1: {
+        nestedField1: ["nested field is not a valid number."]
+      }
+  }
+  ```
+
+This component is configured to work together with a `gpii.schema.parser` and to further evolve the error messages
+where possible.  See [the parser documentation](parser.md) for details.
+
+# Configuration Options
+
+You are expected to configure one or more schemas which will be read from `options.schemaContents`.  The `schemaKey`
+value passed to `validate` is expected to match one of the keys in `options.schemaContent`.
+
+You can change the behavior of AJV by updating `options.validatorOptions` (check their documentation for the syntax).
+
+# Reusing Content between Schemas
+
+One of the key strengths of JSON Schema is that it allows you to compose a complex schema out of parts taken from
+other schemas.  The validator in this package supports references between JSON Schemas.
+
+Inheritance is still a sticking point at least in v4 of the draft standard.   You cannot safely expect to overlay
```

Comment:
Can you give an example here to exhibit what you mean by "reuse individual definitions between schemas"?